Checking a merge of `develop` into `master`

dazza-codes opened the pull request:
This is a check-point to monitor the health of merging `develop` into `master`.

malakai97 committed:
move over api_v1 adapters to api_v2, and test

malakai97 committed:
move over api_v1 controllers to api_v2, and test

malakai97 committed:
offer granularity in bag fabricator

malakai97 committed:
Move over any missed api_v1 things, and remove api_v1

malakai97 committed:
Merge branch 'digest' into develop

malakai97 committed:
migration

malakai97 committed:
migration

malakai97 committed:
model and changes on bag, node

malakai97 committed:
model and changes on bag, node

malakai97 committed:
add copyright statement to migration

malakai97 committed:
add copyright statement to migration

malakai97 committed:
tests and passing

malakai97 committed:
tests and passing

malakai97 committed:
fixity check adapter

malakai97 committed:
fixity check adapter

malakai97 committed:
Merge branch 'master' into develop

Conflicts:
	db/schema.rb

malakai97 committed:
Merge branch 'master' into develop

Conflicts:
	db/schema.rb

malakai97 committed:
Merge branch 'develop' of github.com:dpn-admin/dpn-server into develop

Conflicts:
	app/jobs/bag_man/bag_retrieval_job.rb
	db/schema.rb

malakai97 committed:
Merge branch 'develop' into fixity_check

malakai97 committed:
Merge branch 'fixity_check' of github.com:dpn-admin/dpn-server into fixity_check

malakai97 committed:
fixity checks controller

malakai97 committed:
Merge branch 'fixity_check' into develop

malakai97 committed:
first crack at ingest model

malakai97 committed:
ingest model and specs passing

malakai97 committed:
Ingest adapter with passing specs

malakai97 committed:
ingest controller

malakai97 committed:
Merge pull request #15 from dpn-admin/feature/ingest

Feature/ingest

malakai97 committed:
Discontinue use of the strategy of namespacing non-models
Change routes accordingly.
Version the application via a VERSION variable.

malakai97 committed:
Merge pull request #16 from dpn-admin/remove_namespace_develop

Discontinue use of the strategy of namespacing non-models

malakai97 committed:
Merge branch 'master' into develop

Conflicts:
	app/adapters/api_v1/member_adapter.rb
	app/adapters/bag_adapter.rb
	app/adapters/member_adapter.rb
	app/adapters/node_adapter.rb
	app/adapters/replication_transfer_adapter.rb
	app/adapters/restore_transfer_adapter.rb
	app/controllers/bags_controller.rb
	app/controllers/members_controller.rb
	app/controllers/nodes_controller.rb
	app/controllers/replication_transfers_controller.rb
	app/controllers/restore_transfers_controller.rb
	app/models/fixity_alg.rb
	app/models/fixity_check.rb
	config/initializers/version.rb
	config/routes.rb
	spec/adapters/bag_adapter_spec.rb
	spec/adapters/restore_transfer_adapter_spec.rb
	spec/fabricators/bag_fabricator.rb
	spec/fabricators/fixity_alg_fabricator.rb
	spec/fabricators/node_fabricator.rb
	spec/fabricators/protocol_fabricator.rb
	spec/fabricators/storage_region_fabricator.rb
	spec/fabricators/storage_type_fabricator.rb
	spec/models/bag_spec.rb
	spec/routing/member_routes_spec.rb
	spec/support/examples/a_create_endpoint.rb

malakai97 committed:
Merge branch 'master' into develop

malakai97 committed:
add lograge gem

malakai97 committed:
add lograge gem

malakai97 committed:
use lograge

malakai97 committed:
use lograge

malakai97 committed:
use logstash

malakai97 committed:
use logstash

malakai97 committed:
Set the fractional seconds of timestamps to 0 in an_authorized_update

malakai97 committed:
Move the logic that handles whether or not to honor the updated_at
  field into models (from controllers), and add appropriate tests.
Update bag and node specs to maintain proper setup following this change.

malakai97 committed:
Incorprate the synchronization code from dpn-repl
Add ::find_fields to models
Rewrite sync jobs for dependency injection
Switch from delayed jobs to resque.

malakai97 committed:
Merge branch 'feature/better_logging' of github.com:dpn-admin/dpn-server into feature/better_logging

malakai97 committed:
add Bag#update_with_associations methods, use in controller.

malakai97 committed:
message digests controller should use find_by, not where

malakai97 committed:
add Node#update_with_associations, adjust controller

malakai97 committed:
Convert params configuration in controllers to stateless style.
Standardize on create_params(params) and update_params(params)

malakai97 committed:
Merge pull request #22 from dpn-admin/fixup/models_update

Fixup/models update

malakai97 committed:
Merge branch 'develop' into feature/better_logging

Commit:
Rename uuid to bag in the public hash for restore/replication transfers

Commit:
Update adapter specs to match the updated public hash

Commit:
Update controller specs to replace the correct option in invalid updates

malakai97 committed:
Resolve DPN-212: Add ingest_node parameter to bag index.

malakai97 committed:
Resolve DPN-213: Add total_size field to the result rendered by BagsController#index

malakai97 committed:
Resolve DPN-216: Add ingested parameter to ingests controller index

malakai97 committed:
Resolve DPN-214: Add replicated_by parameter

malakai97 committed:
Add tests for all bag scopes

Commit:
Add success param to fixity_check controller

malakai97 committed:
don't always use true in fixity check adapter

malakai97 committed:
Add map_bool method to adapter, and use on fixity_check, ingest

malakai97 committed:
Resolve DPN-217: Add latest parameter to ingest index

malakai97 committed:
Merge branch 'feature/reporting' of github.com:dpn-admin/dpn-server into feature/reporting

malakai97 committed:
ingests controller should convert 'latest' parameter to boolean

malakai97 committed:
Resolve DPN-215: Add latest parameter to fixity check index

malakai97 committed:
Merge branch 'develop' into feature/better_logging

malakai97 committed:
Make latest_only scopes into single lines, (possibly) addressing a bug in travis.

malakai97 committed:
gitignore .rspec

malakai97 committed:
Refactor ReplicationTransfer
Refactor RestoreTransfer
Remove authz for above based on requester
Replication and Restore transfers no longer mint their own id's
  Use SecureRandom.uuid

refactor replication transfer, restore tranfer, bag man request
don't implicitly mint replication_ids; we don't for anything else
Remove logic from ReplicationTransfer based on requester

malakai97 committed:
Add a gitattributes file

dazza-codes committed:
Travis - report sqlite version

dazza-codes committed:
Travis - use a "ci" rake task

dazza-codes committed:
Bundle update sqlite3

malakai97 committed:
Revert "Travis - use a "ci" rake task"

This reverts commit d072c3f49de3a7d1840736652ae44f872c69a8c3.

malakai97 committed:
Mark tests that fail erroneously in ci with :broken_in_ci
Instruct travis rspec to exclude these tests.

malakai97 committed:
Merge pull request #27 from dpn-admin/feature/reporting

Feature/reporting

malakai97 committed:
Merge branch 'develop' into feature/better_logging

malakai97 committed:
Update to 2.3.1, now include a .ruby_version file

malakai97 committed:
Remove :db time format that broke tests in weird ways.

malakai97 committed:
Merge pull request #28 from dpn-admin/feature/repl

Incorprate the synchronization code from dpn-repl

malakai97 committed:
Merge pull request #29 from dpn-admin/feature/requests

Refactor replication and restore transfers, and update replication code in bag_man_request

malakai97 committed:
Merge branch 'develop' into feature/better_logging

malakai97 committed:
Clarify BagManRequest#staging_location and fix unreliable test.

malakai97 committed:
Fix bug where Sync::Job would send an array of queries to RemoteClient#execute instead of successively sending them.

malakai97 committed:
Resolve DPN-210 and DPN-236:
Create jobs to update the from_node when our ReplicationTransfer changes.
Improve and rename BagMan code to Client::Repl.
Reorganize Sync code into Client::Sync.
Separete secret and non-secret configuration.
Allow configuration to disable the queue adapter, and thus clients.

Commit:
Add yml for ingest, message digest, and fixtity checks

Commit:
Update yml for replication/restore for api-v2 spec

Commit:
Update models in fixture_helper script to match the api-v2 spec

malakai97 committed:
impersonate env defaults to inline queue adapter

malakai97 committed:
Merge pull request #30 from dpn-admin/sync_and_repl

Sync and repl

malakai97 committed:
Change member field "uuid" to "member_id".

malakai97 committed:
Client::Sync::Job must include Client::Common

malakai97 committed:
Update resque.rake schedule to consult the correct jobs.

malakai97 committed:
Merge branch 'develop' into feature/better_logging

malakai97 committed:
Merge pull request #32 from dpn-admin/member_id

Change member field "uuid" to "member_id".

malakai97 committed:
Upgrade gems.
Remove unused gems.
Specify ruby version (2.3.0 or greater) in Gemfile.
Minor updates to specs, models to address warnings from rspec.

malakai97 committed:
Remove RailsAdmin-powered admin interface.

malakai97 committed:
Merge pull request #34 from dpn-admin/remove_admin

Remove RailsAdmin-powered admin interface.

malakai97 committed:
Merge branch 'develop' into feature/better_logging

Conflicts:
	Gemfile
	Gemfile.lock

malakai97 committed:
Merge pull request #35 from dpn-admin/feature/better_logging

Feature/better logging

malakai97 committed:
update readme firstpass

diamondap committed:
Added local YAML and rbenv to gitignore

diamondap committed:
Merge tag 'v2.0.0-rc1' into updated-fixtures

initial v2 release candidate

diamondap committed:
Added impersonate settings to example configs

diamondap committed:
Changed uuid to member_id

diamondap committed:
Updated Cluster doc for 2.0

malakai97 committed:
fix output of rake config:cipher [ci skip]

dazza-codes committed:
db/schema.rb cleanup - it no longer has a delayed_jobs table

malakai97 committed:
Merge pull request #37 from sul-dlss/sdr-api-v2

db/schema.rb cleanup - it no longer has a delayed_jobs table

dazza-codes committed:
script/setup_cluster.rb requires an "impersonate" configuration

dazza-codes committed:
Cluster - add "impersonate" to config template

dazza-codes committed:
Cluster - move configs from .example files into lib/templates/config

dazza-codes committed:
Cluster - update information about using `rake config`

malakai97 committed:
Move ruby version requirement before import of Gemfile.local [ci skip]

malakai97 committed:
Merge pull request #44 from dpn-admin/updated-fixtures

Merge 'updated fixtures' into 'develop'

dazza-codes committed:
Cluster - update template for config/secrets.yml to work with `rake config`

dazza-codes committed:
RAILS_ENV=development -- move some config params to config/secrets.yml

dazza-codes committed:
Merge pull request #45 from dpn-admin/impersonate-secrets

Cluster - update template for config/secrets.yml to work with `rake config`

dazza-codes committed:
Merge pull request #46 from dpn-admin/development-env-use-secrets

RAILS_ENV=development -- move some config params to config/secrets.yml

dazza-codes committed:
Fix impersonate configs

dazza-codes committed:
Fixtures - add a rake task to validate fixture data loaded by script/run_cluster.rb

dazza-codes committed:
Impersonate fixtures - use lower-case for storage region/type name

smutniak committed:
Merge pull request #47 from dpn-admin/impersonate-config-fix

fixed impersonate configs

Commit:
Update Travis secure token

Move over to the Internet2 Slack channel i2-dpn-notices

dazza-codes committed:
MySQL init SQL scripts

malakai97 committed:
Fix bug in migration that breaks mysql [ci skip]

malakai97 committed:
Fix foreign key inconsistencies that broke migrations in at least mysql [ci skip]
  Revert "Fix bug in migration that breaks mysql"
  This reverts commit 87845a1b5c58c42b54e2190231a3e48e73ea4469.

dazza-codes committed:
spec/support/matchers/fuzzy_nested_match.rb - use custom time matcher for +/- 1 sec diffs

dazza-codes committed:
Revise migration for fixity_checks to message_digests

dazza-codes committed:
Revise migration to re-create fixity_checks

dazza-codes committed:
Gemfile.lock - remove mysql2

dazza-codes committed:
db/schema.rb for sqlite -- remove MySQL details

malakai97 committed:
Merge pull request #55 from dpn-admin/develop-remove-mysql

remove mysql details from develop branch

malakai97 committed:
Add tests for Client::Sync::Job#perform, and make green.

malakai97 committed:
LastSuccessManager now correctly creates a default RunTime instance

dazza-codes committed:
Merge pull request #59 from dpn-admin/develop-fix-default_last_success

LastSuccessManager now correctly creates a default RunTime instance

malakai97 committed:
Merge pull request #60 from dpn-admin/spec-support-fuzzy-date-matcher

spec/support/matchers/fuzzy_nested_match.rb - use custom time matcher

malakai97 committed:
use let in job spec

dazza-codes committed:
Merge pull request #57 from dpn-admin/develop-fix-syncjobs

Fix bad initialization in Client::Sync::Job#perform

malakai97 committed:
Merge pull request #54 from dpn-admin/develop-migration-tweaks

db:migrate tweaks that worked for SDR deployment

dazza-codes committed:
Add swagger-ui for /api-docs route

dazza-codes committed:
Gems - add rubocop/rubocop-rspec for development

dazza-codes committed:
config/boot.rb - trivial whitespace change

dazza-codes committed:
Gems - group gems and revise asset gems

dazza-codes committed:
app/views/layouts/application.html.erb - rename app title

dazza-codes committed:
swagger-ui - add symlinks to dist/swagger.* from dpn-rest-spec submodule

malakai97 committed:
Add after_update callback to ReplicationTransfer to checks fixity, requests storage

malakai97 committed:
Merge pull request #68 from dpn-admin/min-changes

Minimal changes extracted from swagger-ui work

dazza-codes committed:
README - add git submodule command to setup instructions

Commit:
Add Dockerfile for this branch. 

Includes the git submodule update

dazza-codes committed:
Merge pull request #67 from dpn-admin/swagger-ui

add swagger-ui for /api-docs

malakai97 committed:
Move the after_update callbacks from ReplicationTransfer to a poro ReplicationTransferUpdater
Use ReplicationTransferUpdater in controller

diamondap committed:
Fixtures: encrypt Node.auth_credential with local cipher key

Commit:
set release tag and delete extraneous content

malakai97 committed:
Fix an issue where Client::Sync::QueryBuilder tests were not being run

Commit:
Merge pull request #73 from dpn-admin/develop-dockerfile-update

set release tag and delete extraneous content

malakai97 committed:
add cancel_reason_detail

malakai97 committed:
Correctly display resque schedule.
Do not attempt to load redis,resque if not in use.

dazza-codes committed:
Merge pull request #74 from dpn-admin/ignored_tests

Fix an issue where Client::Sync::QueryBuilder tests were not being run

malakai97 committed:
Merge pull request #76 from dpn-admin/fix/resque_schedule

Display resque schedule, do not load unused redis

Pcolar commented:
We can overwrite Dockerfile on this merge. I'll check for any issues afterward.  Any changes would be trivial.

dazza-codes committed:
Merge pull request #69 from dpn-admin/auto-store-requested

dpn-server will now automatically check fixity, request storage

dazza-codes committed:
Add simplecov

malakai97 committed:
Merge pull request #82 from dpn-admin/simplecov

Add simplecov

dazza-codes committed:
Add application semantic version with an /api-v2/version route

dazza-codes committed:
Add okcomputer app monitor at /status routes

malakai97 committed:
Merge pull request #85 from dpn-admin/app-monitoring

Add okcomputer app monitor at /status routes

malakai97 committed:
add Node#local_node?

malakai97 committed:
use Node#local_node? in each place it makes sense

dazza-codes committed:
Add migration to ensure MessageDigest.node_id is not null

malakai97 committed:
Merge pull request #84 from dpn-admin/app-sem-version

Add application semantic version with an /api-v2/version route

dazza-codes committed:
Merge pull request #77 from dpn-admin/feature/cancel_reason_detail

Feature/cancel reason detail

malakai97 committed:
Merge pull request #88 from dpn-admin/node-not-null-in-message-digest

Add migration to ensure MessageDigest.node_id is not null

dazza-codes committed:
Merge pull request #89 from dpn-admin/enh/local_node_predicate

Adds Node#local_node?

dazza-codes committed:
specs - clarify coupling of ReplicationTransfer and BagManRequest

BagManRequest specs - clarify behavior of cancel!
ReplicationTransfer specs - clarify behavior of cancel!

malakai97 committed:
Merge pull request #90 from dpn-admin/check-specs

Specs to clarify coupling between ReplicationTransfer and BagManRequest

dazza-codes committed:
Gemfile - move app_version_tasks to dev/test groups

dazza-codes committed:
Merge pull request #91 from dpn-admin/app_version_tasks-dev

Gemfile - move app_version_tasks to dev/test groups

dazza-codes committed:
Rakefile - suppress failure to load app_version_tasks in production

smutniak committed:
Merge pull request #92 from dpn-admin/app_version_tasks-suppress-load-failure

Rakefile - suppress failure to load app_version_tasks in production

dazza-codes committed:
Fix message digests migrations

malakai97 committed:
Fix mysql migration by removing nullify foreign key then re-creating it as restrict

malakai97 committed:
create shared examples for created/updated_before/after specs

malakai97 committed:
use 'it has temporal scopes for' on Bag

malakai97 committed:
Use 'it has temporal scopes for' on Member, add scopes

malakai97 committed:
FixityCheck fabricator sets fixity_at based on created_at if given

malakai97 committed:
use 'it has temporal scopes for' on FixityCheck, fix scopes

malakai97 committed:
use 'it has temporal scopes for' on Ingest, fix scopes

malakai97 committed:
add exclude matcher, the negated form of include

malakai97 committed:
use 'it has temporal scopes for' on MessageDigest, fix scopes

malakai97 committed:
use 'it has temporal scopes for' on Node, add scopes

malakai97 committed:
use 'it has temporal scopes for' on ReplicationTransfer

malakai97 committed:
use 'it has temporal scopes for' on RestoreTransfer

dazza-codes committed:
Merge pull request #104 from dpn-admin/fix/103

Fix mysql migration by removing nullify foreign key then re-creating it as restrict

malakai97 committed:
create shared examples 'a boolean filter' for boolean scopes

malakai97 committed:
test ReplicationTransfer boolean scopes, fix

dazza-codes committed:
Merge pull request #105 from dpn-admin/fix/temporal_scopes

Fix/temporal scopes

malakai97 committed:
test Ingest boolean scope (with_ingested), fix

malakai97 committed:
test RestoreTransfer boolean scopes, fix

dazza-codes committed:
Merge pull request #107 from dpn-admin/fix/boolean_filters

Fix/boolean filters

malakai97 committed:
Merge pull request #106 from dpn-admin/fix-message-digests-migration

Fix message digests migrations

diamondap committed:
Added own-node fixity checks for integration testing

malakai97 committed:
Fixed a bug where adapter#map_bool would coerce nil values to false.

dazza-codes committed:
Merge pull request #115 from dpn-admin/fix/issue_114

Fixed a bug where adapter#map_bool would coerce nil values to false.

malakai97 committed:
add BatchMode=yes, ChallengeResponseAuthentication=no, ConnectTimeout=3 to rsync/ssh command

diamondap committed:
Merge branch 'develop' into fixity-fixtures

malakai97 committed:
Merge pull request #113 from dpn-admin/fixity-fixtures

Added own-node fixity checks for integration testing

malakai97 committed:
Build mysql, sqlite, and postgres in travis

dazza-codes committed:
Postgres init script

malakai97 committed:
Fix bug where postgres adapter would return the total_size sum as a string

malakai97 committed:
Fix refactor replication transfer migration by specifying enum value instead of key

malakai97 committed:
add rollback support for delete_table_delayed_jobs migration

malakai97 committed:
add information to remove_column to enable rollback of change_table_restore_transfers migration

malakai97 committed:
print out versions of databases in travis

malakai97 committed:
Narrow 'broken_in_ci' to mean 'broken_in_sqlite'

dazza-codes committed:
Merge pull request #118 from dpn-admin/multi_db_ci

Build multiple databases in travis ci

dazza-codes committed:
Add scripts to run rspec on MySQL and Postgres

malakai97 committed:
Convert fixity_check#latest_only to SQL conformant code; marked fixed in all database backends

malakai97 committed:
Convert ingest#latest_only to SQL conformant code; marked fixed in all database backends

dazza-codes committed:
Fix message_digests.created_at migration by adding default time

malakai97 committed:
Merge pull request #119 from dpn-admin/develop-migration-fixes

Develop migration fixes

dazza-codes committed:
Merge pull request #123 from dpn-admin/fix_latest_only_scope

Fix latest only scope

malakai97 committed:
Fail on definition if date_format not passed to map_date

malakai97 committed:
memoize rsync options in BagRetrievalJob

dazza-codes committed:
Merge pull request #116 from dpn-admin/ssh_options

Fix bug where ill-configured rsync-over-ssh connections would hang instead of fail

Commit:
Move Docker build to a new repo

Creating dpn-docker repo...

Commit:
Merge pull request #127 from dpn-admin/Docker-patch-2

Move Docker build to a new repo

malakai97 committed:
Move time_formats.rb out of initializers, require it earlier in the init process

dazza-codes committed:
Merge pull request #125 from dpn-admin/fix/issue_124

Fix/issue 124

dazza-codes committed:
Add the application semantic version to the OkComputer status details

dazza-codes committed:
Add redis service on travis.ci

malakai97 committed:
Merge pull request #130 from dpn-admin/okcomputer-app-version

Add the application semantic version to the OkComputer status details

malakai97 committed:
Merge pull request #121 from dpn-admin/migration-scripts

Utility scripts to run specs locally for MySQL and Postgres

malakai97 committed:
script/migrate_cluster.rb now properly sets RAILS_ENV

dazza-codes committed:
Merge pull request #135 from dpn-admin/fix/134/script_migrate_cluster

Correctly set RAILS_ENV in script/migrate_cluster.rb

dazza-codes commented:
The conflict is:

```
git diff config/secrets.yml.example
diff --cc config/secrets.yml.example
index 0d22702,4b85949..0000000
--- a/config/secrets.yml.example
+++ b/config/secrets.yml.example
@@@ -37,8 -26,4 +37,12 @@@ production
    salt: some_salt # generate with rake secret
    cipher_key: some_cipher_key
    cipher_iv: some_cipher_iv
 -  transfer_private_key: /home/dpnuser/.ssh/id-rsa_transfer
++<<<<<<< HEAD
 +  local_namespace: hathi
 +  local_api_root: https://dpn.hathitrust.org
 +  staging_dir: /dpn/staging
 +  repo_dir: /dpn/repo
 +  transfer_private_key: /home/dpnuser/.ssh/id-rsa_transfer
++=======
++  transfer_private_key: /home/dpnuser/.ssh/id-rsa_transfer
++>>>>>>> develop
```

dazza-codes commented:
The merge was easily done when the conflict in `config/secrets.yml.example` was resolved; the result is pushed up to the `master-merge-develop` branch.

malakai97 committed:
Model scopes on associations expect model objects, instead of ids.
Add scope specs for new (as in Bag.new) models.

malakai97 committed:
Adapter map_belongs_to now maps models, not the associated model's id.

malakai97 committed:
fix scopes, fix strong params on BagsController (via removal)

malakai97 committed:
fix scopes, fix strong params on FixityChecksController

malakai97 committed:
fix scopes, fix strong params on ReplicationTransfersController

malakai97 committed:
fix scopes, fix strong params on RestoreTransfersController

malakai97 committed:
fix scopes, fix strong params on IngestsController

malakai97 committed:
fix scopes, fix strong params on MessageDigestsController

malakai97 committed:
validates_associated on Bag model
Bag#update_with_associations now expects model objects, not ids

malakai97 committed:
validates_associated on Ingest

malakai97 committed:
validates_associated on FixityCheck

malakai97 committed:
validates_associated on RestoreTransfer

malakai97 committed:
validates_associated on ReplicationTransfer

malakai97 committed:
Create new automation around seeds.rb,
designed to allow more robust and powerful record creation via rake db:seed
Remove production seeds from dpn-server

malakai97 committed:
Remove uniqueness validations on pseudo-primary keys (e.g. member_id)

malakai97 committed:
Merge pull request #144 from dpn-admin/fix/143/record_uniqueness

Remove uniqueness validations on pseudo-primary keys (e.g. member_id)

malakai97 committed:
Merge pull request #142 from dpn-admin/better_seeds

More robust seed functionality

malakai97 committed:
create and use shared example group for 'with_bag' and similar scopes.

malakai97 committed:
Merge pull request #141 from dpn-admin/fix/137/scopes

Redesign adapters to map belongs_to associations to objects, not their IDs
